src/SConscript

```diff
@@ -30,36 +32,54 @@ def generate_daos_version_header(env):
     out = env.Substfile(tmpl_hdr_in, SUBST_DICT=sd)
     print("generated daos version header file: {}".format(out[0].abspath))
 
+def check_header(env, header):
+    """Check a header file"""
+    result = env.CheckHeader(header)
+    Default(result)
+
 def scons():
     """Execute build"""
 
     Import('env', 'base_env', 'prereqs')
 
     # For Common library (tse.h)
     env.AppendUnique(CPPPATH=[Dir('include/daos').srcnode()])
+    base_env.AppendUnique(CPPPATH=[Dir('include/daos').srcnode()])
     env.Install(os.path.join('$PREFIX', 'include/daos'), 'include/daos/tse.h')
 
-    # Detect if we have explicit_bzero
-    conf = Configure(env)
-    if not conf.CheckFunc('explicit_bzero'):
-        env.Append(CCFLAGS=['-DNEED_EXPLICIT_BZERO'])
-    conf.Finish()
-
     # Generic DAOS includes
     env.AppendUnique(CPPPATH=[Dir('include').srcnode()])
     env.AppendUnique(CPPPATH=[Dir('include')])
     base_env.AppendUnique(CPPPATH=[Dir('include').srcnode()])
     base_env.AppendUnique(CPPPATH=[Dir('include')])
     generate_daos_version_header(env)
+
+    if not env.GetOption('clean') and not env.GetOption('help'):
+        print("Checking local headers can be included")
+        config_env = env.Clone()
+        conf = Configure(config_env)
+        # Detect if we have explicit_bzero
+        if not conf.CheckFunc('explicit_bzero'):
+            env.Append(CCFLAGS=['-DNEED_EXPLICIT_BZERO'])
+            base_env.Append(CCFLAGS=['-DNEED_EXPLICIT_BZERO'])
+        conf.Finish()
+
     for header in HEADERS:
-        env.Install(os.path.join('$PREFIX', 'include'), 'include/%s' % header)
+        header_file = "include/%s" % header
+        env.Install(os.path.join('$PREFIX', 'include'), header_file)
     if prereqs.server_requested():
         for header in HEADERS_SRV:
-            env.Install(os.path.join('$PREFIX', 'include', 'daos_srv'),
-                        'include/daos_srv/%s' % header)
+            header_file = "include/daos_srv/%s" % header
+            env.Install(os.path.join('$PREFIX', 'include', 'daos_srv'), header_file)
     for header in HEADERS_GURT:
-        env.Install(os.path.join('$PREFIX', 'include', 'gurt'),
-                    'include/gurt/%s' % header)
+        header_file = "include/gurt/%s" % header
+        check_header(env, header_file)
+        env.Install(os.path.join('$PREFIX', 'include', 'gurt'), header_file)
```

Comment:
How come Default is in the check_header() function, but env.Install is outside of it?

Reply:
Default is trying to force it to do the header check build even if it's not explicitly requested since nothing depends on those files being built in this way.  Install is a separate thing.   But if we want to move it to check_header, that is fine, just need to potentially rename that function to something else.